src/tests/system/tests/test_ipa.py

```diff
@@ -126,6 +129,78 @@ def test_ipa__hostpublickeys_by_ip(client: Client, ipa: IPA, public_keys: list[s
         assert f"{ip} {key}" in result.stdout_lines, "Did not get expected public keys!"
 
 
+@pytest.mark.ticket(gh=7583)
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__hostpublickeys_by_name_with_port(client: Client, ipa: IPA, public_keys: list[str]):
+    """
+    :title: sss_ssh_knownhosts returns public keys by host name with port
+    :setup:
+        1. Create host with SSH key
+        2. Configure SSSD with SSH responder
+        3. Start SSSD
+    :steps:
+        1. Lookup SSH key
+    :expectedresults:
+        1. All public keys were printed
+    :customerscenario: False
+    """
+    hostname = f"ssh-host.{ipa.domain}"
+    ip = "10.255.251.10"
+
+    ipa.host_account(hostname).add(ip=ip, sshpubkey=public_keys)
+    client.sssd.enable_responder("ssh")
+    client.sssd.start()
+    port = 3333
+    result = client.sss_ssh_knownhosts(f"{hostname}:{port}")
+    assert result.rc == 0, "Did not get OpenSSH known hosts public keys!"
+    assert len(public_keys) == len(result.stdout_lines), "Did not get expected number of public keys!"
+    for key in public_keys:
+        assert f"{hostname}:{port} {key}" in result.stdout_lines, ("Did not get expected public keys "
+                                                                   "with the host name with port")
+
+
+@pytest.mark.ticket(gh=7583)
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__hostpublickeys_with_non_default_port(client: Client, ipa: IPA, public_keys: list[str]):
+    """
+    :title: sss_ssh_knownhosts returns public keys by hostname with non-default port
+    :setup:
+        1. Create host with SSH key
+        2. Add the ipasshpubkey with hostname and port
+        3. Configure SSSD with SSH responder
+        4. Start SSSD
+    :steps:
+        1. Lookup SSH key
+    :expectedresults:
+        1. All public keys were printed
+    :customerscenario: False
+    """
+    hostname = f"ssh-host.{ipa.domain}"
+    ip = "10.255.251.10"
+
+    ipa.host_account(hostname).add(ip=ip, sshpubkey=public_keys)
+    client.sssd.enable_responder("ssh")
+    client.sssd.start()
+    # IPA doesn't currently ipa host-mod with hostname and key
+    # this is workaround till IPA add the support.
+    for key in public_keys:
+        modify_content = ipa.fs.mktmp(rf"""
+                        dn: fqdn={hostname},cn=computers,cn=accounts,dc=ipa,dc=test
+                        changetype: modify
+                        add: ipaSshPubKey
+                        ipaSshPubKey: [{hostname}]:4444 {key}
```

Comment:
It would be better to set a port variable and use it instead of hardcoding the port number. As you did in the previous test.

Reply:
updated.